### PR TITLE
Update the azurerm version contraint from 3.105 to 5.0

### LINKED ISSUE #79: [AVM Module Issue]: Update the azurerm version contraint from 3.105 to 5.0
### Check for previous/existing GitHub issues

- [x] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### (Optional) Module Version

_No response_

### (Optional) Correlation Id

_No response_

### Description

Update the azurerm version contraint from 3.105 to 5.0

### PR DESCRIPTION
**Pull request by JamesDLD**
## Description
Update the azurerm version contraint from 3.105 to 5.0

## Type of Change

- [ ] Non-module change (e.g. CI/CD, documentation, etc.)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `locals.version.tf.json`:
    - [x] Someone has opened a bug report issue, and I have included "Closes #79" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `locals.version.tf.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `locals.version.tf.json`.
  - [ ] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] I did run all  [pre-commit](https://azure.github.io/Azure-Verified-Modules/contributing/terraform/terraform-contribution-flow/#5-run-pre-commit-checks) checks

### CODE REVIEW

**Comment by JamesDLD** [comment]
avm pre-commit done 
avm pr-check has the following error i'm not able to understand:
```
------------------------------------------------

The preceding files contain terraform blocks that do not comply with tflint requirements.

make: *** [avmmakefile:43: tflint] Error 1
```

**Comment by jaredfholgate** [comment]
> avm pre-commit done avm pr-check has the following error i'm not able to understand:
> 
> ```
> ------------------------------------------------
> 
> The preceding files contain terraform blocks that do not comply with tflint requirements.
> 
> make: *** [avmmakefile:43: tflint] Error 1
> ```

Can you share the whole output from that command? In the meantime I will run it in the PR too and see if I can see the issue.

**Comment by jaredfholgate** [comment]
> > avm pre-commit done avm pr-check has the following error i'm not able to understand:
> > ```
> > ------------------------------------------------
> > 
> > The preceding files contain terraform blocks that do not comply with tflint requirements.
> > 
> > make: *** [avmmakefile:43: tflint] Error 1
> > ```
> 
> Can you share the whole output from that command? In the meantime I will run it in the PR too and see if I can see the issue.

I see the error in the log:

```
Error: `modtm` provider should be declared in the `required_providers` block (provider_modtm_version)

  on terraform.tf line 1:
   1: terraform {
```

**Comment by JamesDLD** [comment]
Thank you @jaredfholgate, have the following error now even when the provider 'modtm' is not used by the module.
When I declare it as a required module it prints the following message --> provider 'modtm' is declared in required_providers but not used by the module.

```
1 issue(s) found:

Error: `modtm` provider should be declared in the `required_providers` block (provider_modtm_version)

  on terraform.tf line 1:
   1: terraform {

------------------------------------------------

The root module . contains terraform blocks that do not comply with tflint requirements.

==> Checking that sub modules comply with tflint requirements...
==> Checking that examples comply with tflint requirements...
==> Running tflint for example ./basic
Plugin "terraform" is already installed
Plugin "avm" is already installed
==> Running tflint for example ./developer
Plugin "terraform" is already installed
Plugin "avm" is already installed
==> Running tflint for example ./default
Plugin "terraform" is already installed
Plugin "avm" is already installed
------------------------------------------------

The preceding files contain terraform blocks that do not comply with tflint requirements.

make: *** [avmmakefile:43: tflint] Error 1
```

File: examples/default/main.tf [hunk 1]
```diff
@@ -3,7 +3,7 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "~> 3.105"
+      version = "< 5.0"
```
[the remark this answers is not in the document]
Comment (jaredfholgate) [reply examples/default/main.tf]: Still need the test for v3

**Comment by jaredfholgate** [comment]
> Thank you @jaredfholgate, have the following error now even when the provider 'modtm' is not used by the module. When I declare it as a required module it prints the following message --> provider 'modtm' is declared in required_providers but not used by the module.
> 
> ```
> 1 issue(s) found:
> 
> Error: `modtm` provider should be declared in the `required_providers` block (provider_modtm_version)
> 
>   on terraform.tf line 1:
>    1: terraform {
> 
> ------------------------------------------------
> 
> The root module . contains terraform blocks that do not comply with tflint requirements.
> 
> ==> Checking that sub modules comply with tflint requirements...
> ==> Checking that examples comply with tflint requirements...
> ==> Running tflint for example ./basic
> Plugin "terraform" is already installed
> Plugin "avm" is already installed
> ==> Running tflint for example ./developer
> Plugin "terraform" is already installed
> Plugin "avm" is already installed
> ==> Running tflint for example ./default
> Plugin "terraform" is already installed
> Plugin "avm" is already installed
> ------------------------------------------------
> 
> The preceding files contain terraform blocks that do not comply with tflint requirements.
> 
> make: *** [avmmakefile:43: tflint] Error 1
> ```

Apologies for the delay. With customers all week. Will try to fix this ASAP if you don't figure it out in the meantime.

**Comment by JamesDLD** [comment]
Hi @jaredfholgate I think we are good to go now, the pre commits checks are ok, to solve the latest issue I had to update the provider to include modtm for telemtry usage.

**Comment by jaredfholgate** [comment]
Move to a release branch to fix issues and run e2e tests. New PR here: https://github.com/Azure/terraform-azurerm-avm-res-network-bastionhost/pull/81

**Pull request merged**